data/model/modelutil.py

```diff
@@ -69,24 +77,33 @@ def pagination_start(page_token=None):
     return None
 
 
-def paginate_query(query, limit=50, sort_field_name=None, page_number=None):
+def paginate_query(query, limit=50, sort_field_name=None, page_number=None, offset_val=0):
     """
     Executes the given query and returns a page's worth of results, as well as the page token for
     the next page (if any).
     """
     results = list(query)
     page_token = None
+    use_offset = False
     if len(results) > limit:
         start_index = getattr(results[limit], sort_field_name or "id")
         is_datetime = False
         if isinstance(start_index, datetime):
             start_index = start_index.isoformat() + "Z"
+            # datetime does not have a unique constraint, so multiple entries can have the same datetime.
+            # using offset to fetch next page items with same datetime.
+            use_offset = getattr(results[limit], sort_field_name) == getattr(
+                results[limit - 1], sort_field_name
+            )
+            offset_val = limit + offset_val if use_offset else 0
             is_datetime = True
 
         page_token = {
             "start_index": start_index,
             "page_number": page_number + 1 if page_number else 1,
             "is_datetime": is_datetime,
+            "use_offset": use_offset,
```

Comment:
Do you really need `use_offset`, or just to set `offset_val`?

Reply:
nice, I can just use `offset_val `